tel2puml/pipelines/data_creation.py:6: error: Skipping analyzing "test_harness.protocol_verifier.simulator_data": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/pipelines/data_creation.py:10: error: Skipping analyzing "test_event_generator.io.run": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/pipelines/data_creation.py:19: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:52: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:106: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:107: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:119: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:148: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/data_creation.py:161: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/utils.py:13: error: Missing type parameters for generic type "set"  [type-arg]
tel2puml/utils.py:13: error: Missing type parameters for generic type "frozenset"  [type-arg]
tel2puml/utils.py:14: error: Missing type parameters for generic type "set"  [type-arg]
tel2puml/utils.py:15: error: Missing type parameters for generic type "set"  [type-arg]
tel2puml/utils.py:47: error: Incompatible types in assignment (expression has type "set[Any] | frozenset[Any]", variable has type "set[set[Any] | frozenset[Any]]")  [assignment]
tel2puml/utils.py:77: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/utils.py:77: error: Missing type parameters for generic type "tuple"  [type-arg]
tel2puml/utils.py:78: error: Name "plt.Figure" is not defined  [name-defined]
tel2puml/utils.py:108: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/utils.py:142: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/read_uml_file.py:32: error: Skipping analyzing "test_event_generator.solutions.graph_solution": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/read_uml_file.py:36: error: Skipping analyzing "test_event_generator.io.run": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/read_uml_file.py:42: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/read_uml_file.py:43: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/read_uml_file.py:118: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/read_uml_file.py:161: error: Function is missing a type annotation  [no-untyped-def]
tel2puml/read_uml_file.py:203: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/read_uml_file.py:233: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/read_uml_file.py:233: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/read_uml_file.py:258: error: Call to untyped function "stringify_events" in typed context  [no-untyped-call]
tel2puml/read_uml_file.py:292: error: Incompatible types in assignment (expression has type "dict[Never, Never]", target has type "str")  [assignment]
tel2puml/read_uml_file.py:296: error: Unsupported target for indexed assignment ("str")  [index]
tel2puml/read_uml_file.py:303: error: Incompatible types in assignment (expression has type "str", variable has type "PVEvent")  [assignment]
tel2puml/read_uml_file.py:304: error: Invalid index type "PVEvent" for "dict[str, dict[str, str]]"; expected type "str"  [index]
tel2puml/read_uml_file.py:305: error: Invalid index type "PVEvent" for "dict[str, dict[str, str]]"; expected type "str"  [index]
tel2puml/read_uml_file.py:306: error: Invalid index type "PVEvent" for "dict[str, dict[str, str]]"; expected type "str"  [index]
tel2puml/read_uml_file.py:307: error: Unsupported target for indexed assignment ("str")  [index]
tel2puml/read_uml_file.py:307: error: Invalid index type "PVEvent" for "dict[str, dict[str, str]]"; expected type "str"  [index]
tel2puml/read_uml_file.py:309: error: Invalid index type "PVEvent" for "dict[str, dict[str, str]]"; expected type "str"  [index]
tel2puml/read_uml_file.py:309: error: Incompatible types in assignment (expression has type "dict[str, str]", target has type "str")  [assignment]
tel2puml/read_uml_file.py:310: error: Invalid index type "PVEvent" for "dict[str, dict[str, str]]"; expected type "str"  [index]
tel2puml/read_uml_file.py:312: error: Incompatible types in "yield" (actual type "dict[str, dict[str, str]]", expected type "dict[str, NestedEvent]")  [misc]
tel2puml/read_uml_file.py:343: error: Argument 1 to "recursive_get_type" has incompatible type "NestedEvent"; expected "dict[Any, Any]"  [arg-type]
tel2puml/read_uml_file.py:385: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/read_uml_file.py:396: error: Need type annotation for "graph"  [var-annotated]
tel2puml/read_uml_file.py:412: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/read_uml_file.py:413: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/read_uml_file.py:422: error: Non-overlapping equality check (left operand type: "DiDegreeView[Any]", right operand type: "Literal[0]")  [comparison-overlap]
tel2puml/read_uml_file.py:424: error: Non-overlapping equality check (left operand type: "DiDegreeView[Any]", right operand type: "Literal[0]")  [comparison-overlap]
tel2puml/read_uml_file.py:431: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/jAlergia2NetworkX.py:25: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/jAlergia2NetworkX.py:66: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/jAlergia2NetworkX.py:94: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/jAlergia2NetworkX.py:94: note: "convert_to_networkx" defined here
tel2puml/jAlergia2NetworkX.py:95: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/jAlergia2NetworkX.py:110: error: Need type annotation for "graph"  [var-annotated]
tel2puml/jAlergia2NetworkX.py:148: error: Unexpected keyword argument "write" for "convert_to_networkx"  [call-arg]
tel2puml/detect_loops.py:131: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/detect_loops.py:147: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tel2puml/detect_loops.py:296: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/detect_loops.py:314: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[str]")  [assignment]
tel2puml/check_puml_equiv.py:11: error: Skipping analyzing "test_event_generator.io.parse_puml": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/check_puml_equiv.py:53: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/check_puml_equiv.py:111: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:124: error: Need type annotation for "graph"  [var-annotated]
tel2puml/check_puml_equiv.py:129: error: Need type annotation for "collected_attributes" (hint: "collected_attributes: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/check_puml_equiv.py:219: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:228: error: Argument 1 to "create_networkx_graph_from_parsed_puml" has incompatible type "list[str]"; expected "list[Any | tuple[Literal['START', 'END', 'PATH'], Literal['XOR', 'AND', 'OR', 'LOOP']]]"  [arg-type]
tel2puml/check_puml_equiv.py:228: note: "List" is invariant -- see https://mypy.readthedocs.io/en/stable/common_issues.html#variance
tel2puml/check_puml_equiv.py:228: note: Consider using "Sequence" instead, which is covariant
tel2puml/check_puml_equiv.py:269: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/check_puml_equiv.py:280: error: Returning Any from function declared to return "bool"  [no-any-return]
tests/test_utils.py:9: error: Need type annotation for "graph"  [var-annotated]
tests/test_read_uml_file.py:132: error: Argument 1 to "get_markov_sequence_lines_from_audit_event_stream" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_read_uml_file.py:139: error: Argument 1 to "get_markov_sequence_lines_from_audit_event_stream" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_read_uml_file.py:150: error: Argument 1 to "get_markov_sequence_lines_from_audit_event_stream" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_check_puml_equiv.py:5: error: Skipping analyzing "test_event_generator.io.parse_puml": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tests/test_check_puml_equiv.py:33: error: Need type annotation for "collected_attributes" (hint: "collected_attributes: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_check_puml_equiv.py:104: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tests/test_check_puml_equiv.py:213: error: Need type annotation for "graph"  [var-annotated]
tests/test_check_puml_equiv.py:217: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:218: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:221: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:222: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:253: error: Need type annotation for "graph"  [var-annotated]
tests/test_check_puml_equiv.py:257: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:258: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:261: error: Dict entry 0 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:262: error: Dict entry 1 has incompatible type "str": "bool"; expected "str": "str"  [dict-item]
tests/test_check_puml_equiv.py:285: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
tests/test_check_puml_equiv.py:286: error: Incompatible types in assignment (expression has type "bool", target has type "str")  [assignment]
tel2puml/jAlergiaPipeline.py:22: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/jAlergiaPipeline.py:39: error: Incompatible types in assignment (expression has type "list[str]", variable has type "str")  [assignment]
tel2puml/jAlergiaPipeline.py:55: error: Incompatible types in assignment (expression has type "list[str]", variable has type "str")  [assignment]
tel2puml/jAlergiaPipeline.py:98: error: Missing type parameters for generic type "MultiDiGraph"  [type-arg]
tel2puml/jAlergiaPipeline.py:98: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/jAlergiaPipeline.py:120: error: Missing type parameters for generic type "MultiDiGraph"  [type-arg]
tel2puml/jAlergiaPipeline.py:120: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/jAlergiaPipeline.py:142: error: Missing type parameters for generic type "MultiDiGraph"  [type-arg]
tel2puml/jAlergiaPipeline.py:159: error: Missing type parameters for generic type "MultiDiGraph"  [type-arg]
tel2puml/puml_graph/graph.py:55: error: "type[PUMLOperatorNodes]" has no attribute "PATH_LOOP"  [attr-defined]
tel2puml/puml_graph/graph.py:78: error: Argument "extra_info" to "__init__" of "NXNode" has incompatible type "dict[str, bool] | None"; expected "dict[str, str] | None"  [arg-type]
tel2puml/puml_graph/graph.py:296: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/puml_graph/graph.py:390: error: Incompatible return value type (got "tuple[PUMLOperatorNode, ...]", expected "tuple[PUMLOperatorNode, PUMLOperatorNode]")  [return-value]
tel2puml/puml_graph/graph.py:438: error: Signature of "add_node" incompatible with supertype "Graph"  [override]
tel2puml/puml_graph/graph.py:438: note:      Superclass:
tel2puml/puml_graph/graph.py:438: note:          def add_node(self, node_for_adding: Any, **attr: Any) -> None
tel2puml/puml_graph/graph.py:438: note:      Subclass:
tel2puml/puml_graph/graph.py:438: note:          def add_node(self, node: PUMLNode) -> None
tel2puml/puml_graph/graph.py:449: error: Signature of "add_edge" incompatible with supertype "Graph"  [override]
tel2puml/puml_graph/graph.py:449: note:      Superclass:
tel2puml/puml_graph/graph.py:449: note:          def add_edge(self, u_of_edge: Any, v_of_edge: Any, **attr: Any) -> None
tel2puml/puml_graph/graph.py:449: note:      Subclass:
tel2puml/puml_graph/graph.py:449: note:          def add_edge(self, start_node: PUMLNode, end_node: PUMLNode) -> None
tel2puml/puml_graph/graph.py:458: error: Incompatible types (expression has type "dict[str, Any]", TypedDict item "start_node_attr" has type "NXNodeAttributes")  [typeddict-item]
tel2puml/puml_graph/graph.py:459: error: Incompatible types (expression has type "dict[str, Any]", TypedDict item "end_node_attr" has type "NXNodeAttributes")  [typeddict-item]
tel2puml/puml_graph/graph.py:533: error: Invalid index type "PUMLNode" for "dict[PUMLEventNode | PUMLOperatorNode, list[PUMLEventNode | PUMLOperatorNode]]"; expected type "PUMLEventNode | PUMLOperatorNode"  [index]
tel2puml/puml_graph/graph.py:539: error: Call to untyped function (unknown) in typed context  [no-untyped-call]
tel2puml/puml_graph/graph.py:551: error: Incompatible return value type (got "list[PUMLNode]", expected "list[PUMLEventNode | PUMLOperatorNode]")  [return-value]
tel2puml/puml_graph/graph.py:642: error: "DiGraph[Any]" has no attribute "node_counts"  [attr-defined]
tel2puml/puml_graph/graph.py:643: error: "DiGraph[Any]" has no attribute "branch_counts"  [attr-defined]
tel2puml/puml_graph/graph.py:644: error: "DiGraph[Any]" has no attribute "kill_counts"  [attr-defined]
tel2puml/puml_graph/graph.py:645: error: "DiGraph[Any]" has no attribute "parent_graph_nodes_to_node_ref"  [attr-defined]
tel2puml/puml_graph/graph.py:648: error: Incompatible return value type (got "DiGraph[Any]", expected "PUMLGraph")  [return-value]
tests/test_jAlergiaPipeline.py:79: error: Missing type parameters for generic type "MultiDiGraph"  [type-arg]
tests/test_jAlergiaPipeline.py:79: error: Missing type parameters for generic type "dict"  [type-arg]
tests/test_jAlergiaPipeline.py:108: error: Argument 1 to "get_markov_sequence_lines_from_audit_event_stream" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_jAlergiaPipeline.py:119: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_jAlergiaPipeline.py:125: error: Missing type parameters for generic type "MultiDiGraph"  [type-arg]
tests/test_jAlergiaPipeline.py:132: error: Need type annotation for "graph"  [var-annotated]
tests/test_jAlergiaPipeline.py:171: error: Need type annotation for "graph"  [var-annotated]
tests/test_detect_loops.py:383: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tests/test_detect_loops.py:404: error: Incompatible types in assignment (expression has type "list[Loop] | list[str]", variable has type "list[Loop]")  [assignment]
tests/test_detect_loops.py:446: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:472: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:491: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:511: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:534: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:581: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:609: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_detect_loops.py:644: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/puml_graph/test_graph.py:5: error: Skipping analyzing "test_event_generator.io.parse_puml": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tests/puml_graph/test_graph.py:47: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/puml_graph/test_graph.py:311: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/puml_graph/test_graph.py:320: error: Non-overlapping equality check (left operand type: "dict[str, str]", right operand type: "dict[str, bool]")  [comparison-overlap]
tests/puml_graph/test_graph.py:533: error: Need type annotation for "node_set" (hint: "node_set: Set[<type>] = ...")  [var-annotated]
tests/puml_graph/test_graph.py:579: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:580: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:581: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:606: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:608: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:609: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:611: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:613: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:639: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:641: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:642: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph.py:644: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tests/puml_graph/test_graph.py:646: error: Item "None" of "PUMLGraph | None" has no attribute "edges"  [union-attr]
tel2puml/puml_graph/graph_loop_extract.py:56: error: Need type annotation for "start_loop_nodes" (hint: "start_loop_nodes: Set[<type>] = ...")  [var-annotated]
tel2puml/puml_graph/graph_loop_extract.py:57: error: Need type annotation for "end_loop_nodes" (hint: "end_loop_nodes: Set[<type>] = ...")  [var-annotated]
tests/puml_graph/test_graph_loop_extract.py:57: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:63: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:74: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:80: error: Argument 1 to "remove" of "list" has incompatible type "Hashable"; expected "str"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:302: error: Argument 1 to "remove" of "list" has incompatible type "tuple[Hashable, Hashable]"; expected "tuple[tuple[str, str, int], tuple[str, str, int]]"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:320: error: Argument 1 to "check_correct_start_and_end_nodes" of "TestGetLoopStartAndEnd" has incompatible type "list[tuple[PUMLNode, PUMLNode]]"; expected "list[tuple[PUMLEventNode, PUMLEventNode]]"  [arg-type]
tests/puml_graph/test_graph_loop_extract.py:335: error: Argument 1 to "check_correct_start_and_end_nodes" of "TestGetLoopStartAndEnd" has incompatible type "list[tuple[PUMLNode, PUMLNode]]"; expected "list[tuple[PUMLEventNode, PUMLEventNode]]"  [arg-type]
tel2puml/puml_graph/graph_loop_insert.py:42: error: Argument 1 to "insert_loops" has incompatible type "PUMLGraph | None"; expected "PUMLGraph"  [arg-type]
tests/puml_graph/test_graph_loop_insert.py:47: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:57: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:73: error: Item "None" of "PUMLGraph | None" has no attribute "nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:79: error: Item "None" of "PUMLGraph | None" has no attribute "subgraph_nodes"  [union-attr]
tests/puml_graph/test_graph_loop_insert.py:95: error: Argument 1 to "check_networkx_graph_equivalence" has incompatible type "PUMLGraph | None"; expected "DiGraph[Any]"  [arg-type]
tests/puml_graph/test_graph_loop_insert.py:100: error: Argument 1 to "check_networkx_graph_equivalence" has incompatible type "PUMLGraph | None"; expected "DiGraph[Any]"  [arg-type]
tel2puml/pipelines/logic_detection_pipeline.py:14: error: Skipping analyzing "pm4py": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/pipelines/logic_detection_pipeline.py:21: error: Skipping analyzing "test_event_generator.solutions.graph_solution": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/pipelines/logic_detection_pipeline.py:22: error: Skipping analyzing "test_event_generator.solutions.event_solution": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/pipelines/logic_detection_pipeline.py:50: error: Function is missing a type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:59: error: Function is missing a type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:100: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:103: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:104: error: Call to untyped function "__key" in typed context  [no-untyped-call]
tel2puml/pipelines/logic_detection_pipeline.py:106: error: Function is missing a type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:107: error: Call to untyped function "__key" in typed context  [no-untyped-call]
tel2puml/pipelines/logic_detection_pipeline.py:144: error: Missing type parameters for generic type "ndarray"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:145: error: Missing type parameters for generic type "ndarray"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:583: error: Argument 1 to "remove" of "set" has incompatible type "set[Any]"; expected "frozenset[str]"  [arg-type]
tel2puml/pipelines/logic_detection_pipeline.py:586: error: Argument 1 to "get_weighted_cover" has incompatible type "set[frozenset[str]]"; expected "set[set[Any] | frozenset[Any]]"  [arg-type]
tel2puml/pipelines/logic_detection_pipeline.py:643: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:664: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/pipelines/logic_detection_pipeline.py:693: error: Need type annotation for "data_point_edges" (hint: "data_point_edges: Set[<type>] = ...")  [var-annotated]
tel2puml/pipelines/logic_detection_pipeline.py:786: error: Value of type "ndarray[Any, Any] | None" is not indexable  [index]
tel2puml/pipelines/logic_detection_pipeline.py:786: error: Unsupported target for indexed assignment ("ndarray[Any, Any] | None")  [index]
tel2puml/pipelines/logic_detection_pipeline.py:791: error: Missing type parameters for generic type "ndarray"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:806: error: Missing type parameters for generic type "ndarray"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:812: error: Returning Any from function declared to return "ndarray[Any, Any]"  [no-any-return]
tel2puml/pipelines/logic_detection_pipeline.py:813: error: Argument 1 to "diag" has incompatible type "ndarray[Any, Any] | None"; expected "_SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]]"  [arg-type]
tel2puml/pipelines/logic_detection_pipeline.py:820: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:832: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "dict_values[str, list[dict[Any, Any]]]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tel2puml/pipelines/logic_detection_pipeline.py:927: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:941: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:947: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:1067: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/pipelines/logic_detection_pipeline.py:1076: error: Need type annotation for "graph"  [var-annotated]
tel2puml/pipelines/logic_detection_pipeline.py:1098: error: Value expression in dictionary comprehension has incompatible type "str"; expected type "Event"  [misc]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:4: error: Skipping analyzing "pm4py": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:111: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:113: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:115: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:120: error: Unsupported target for indexed assignment ("ndarray[Any, Any] | None")  [index]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:121: error: Unsupported target for indexed assignment ("ndarray[Any, Any] | None")  [index]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:123: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:149: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:166: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:173: error: Need type annotation for "data_point_edges" (hint: "data_point_edges: Set[<type>] = ...")  [var-annotated]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:180: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:190: error: Item "None" of "ndarray[Any, Any] | None" has no attribute "tolist"  [union-attr]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:207: error: "None" has no attribute "tolist"  [attr-defined]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:213: error: "None" has no attribute "tolist"  [attr-defined]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:219: error: "None" has no attribute "tolist"  [attr-defined]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:363: error: Argument 1 to "create_augmented_data_from_reduced_event_set" of "Event" has incompatible type "EventSet"; expected "frozenset[str]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:523: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:542: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:558: error: Call to untyped function "_check_or" in typed context  [no-untyped-call]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:565: error: Call to untyped function "_check_or_and" in typed context  [no-untyped-call]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:759: error: Function is missing a return type annotation  [no-untyped-def]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:851: error: Function is missing a type annotation  [no-untyped-def]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:860: error: Call to untyped function "_check_node_for_repeat_and" in typed context  [no-untyped-call]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:880: error: Call to untyped function "_check_node_for_repeat_and" in typed context  [no-untyped-call]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1036: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1104: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1143: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1193: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1241: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1291: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1345: error: Argument 1 to "update_all_connections_from_data" has incompatible type "Generator[dict[Any, Any], Any, None]"; expected "list[dict[Any, Any]]"  [arg-type]
tests/tel2puml/pipelines/test_logic_detection_pipeline.py:1460: error: Argument 1 to "get_graph_solutions_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tel2puml/node_map_to_puml/node.py:10: error: Skipping analyzing "pm4py": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/node_map_to_puml/node.py:11: error: Skipping analyzing "pm4py.objects.process_tree.obj": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/node_map_to_puml/node.py:11: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
tel2puml/node_map_to_puml/node.py:89: error: Unsupported left operand type for + ("None")  [operator]
tel2puml/node_map_to_puml/node.py:89: note: Left operand is of type "str | None"
tel2puml/node_map_to_puml/node.py:259: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node.py:261: error: Incompatible default for argument "uid" (default has type "None", argument has type "str")  [assignment]
tel2puml/node_map_to_puml/node.py:261: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:261: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:262: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:262: error: Incompatible default for argument "incoming" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:262: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:262: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:263: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:263: error: Incompatible default for argument "outgoing" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:263: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:263: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:264: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:264: error: Incompatible default for argument "incoming_logic" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:264: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:264: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:265: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node.py:265: error: Incompatible default for argument "outgoing_logic" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node.py:265: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node.py:265: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node.py:329: error: Incompatible return value type (got "tuple[PUMLEvent, ...]", expected "tuple[PUMLEvent] | None")  [return-value]
tel2puml/node_map_to_puml/node.py:416: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:418: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:429: error: Need type annotation for "networkx_graph"  [var-annotated]
tel2puml/node_map_to_puml/node.py:454: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:463: error: Need type annotation for "event_node_ref" (hint: "event_node_ref: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/node_map_to_puml/node.py:476: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:479: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:554: error: No overload variant of "__setitem__" of "list" matches argument types "int", "PUMLNode"  [call-overload]
tel2puml/node_map_to_puml/node.py:554: note: Possible overload variants:
tel2puml/node_map_to_puml/node.py:554: note:     def __setitem__(self, SupportsIndex, PUMLOperatorNode, /) -> None
tel2puml/node_map_to_puml/node.py:554: note:     def __setitem__(self, slice, Iterable[PUMLOperatorNode], /) -> None
tel2puml/node_map_to_puml/node.py:581: error: Incompatible types in assignment (expression has type "list[PUMLOperatorNode | PUMLNode]", variable has type "list[PUMLOperatorNode]")  [assignment]
tel2puml/node_map_to_puml/node.py:584: error: Incompatible return value type (got "tuple[PUMLNode, Node | None]", expected "tuple[PUMLOperatorNode | PUMLNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:607: error: No overload variant of "__setitem__" of "list" matches argument types "int", "Node"  [call-overload]
tel2puml/node_map_to_puml/node.py:607: note: Possible overload variants:
tel2puml/node_map_to_puml/node.py:607: note:     def __setitem__(self, SupportsIndex, None, /) -> None
tel2puml/node_map_to_puml/node.py:607: note:     def __setitem__(self, slice, Iterable[None], /) -> None
tel2puml/node_map_to_puml/node.py:683: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:695: error: Argument 1 to "create_event_node" of "PUMLGraph" has incompatible type "str | None"; expected "str"  [arg-type]
tel2puml/node_map_to_puml/node.py:722: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLEventNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:737: error: Incompatible types in assignment (expression has type "PUMLOperatorNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:751: error: Argument 2 to "update_puml_graph_with_event_node" has incompatible type "Node | None"; expected "Node"  [arg-type]
tel2puml/node_map_to_puml/node.py:757: error: Incompatible types in assignment (expression has type "PUMLOperatorNode", variable has type "PUMLEventNode")  [assignment]
tel2puml/node_map_to_puml/node.py:792: error: Argument 1 to "create_operator_node_pair" of "PUMLGraph" has incompatible type "PUMLOperator | None"; expected "PUMLOperator"  [arg-type]
tel2puml/node_map_to_puml/node.py:802: error: Incompatible return value type (got "tuple[PUMLOperatorNode | PUMLEventNode, Node]", expected "tuple[PUMLOperatorNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:851: error: Argument 2 to "create_logic_merge" of "LogicBlockHolder" has incompatible type "None"; expected "Node"  [arg-type]
tel2puml/node_map_to_puml/node.py:859: error: Incompatible return value type (got "tuple[PUMLOperatorNode | PUMLEventNode, Node]", expected "tuple[PUMLOperatorNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:866: error: Argument 3 to "handle_rotate_path" has incompatible type "PUMLNode"; expected "PUMLOperatorNode"  [arg-type]
tel2puml/node_map_to_puml/node.py:868: error: Incompatible return value type (got "tuple[PUMLOperatorNode | PUMLEventNode, Node]", expected "tuple[PUMLOperatorNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:893: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/node_map_to_puml/node.py:900: error: Incompatible types in assignment (expression has type "PUMLOperatorNode | PUMLEventNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/node_map_to_puml/node.py:960: error: Incompatible return value type (got "tuple[PUMLNode, Node]", expected "tuple[PUMLOperatorNode | PUMLEventNode, Node]")  [return-value]
tel2puml/node_map_to_puml/node.py:1000: error: Incompatible types in assignment (expression has type "PUMLEventNode", variable has type "PUMLOperatorNode")  [assignment]
tel2puml/node_map_to_puml/node.py:1061: error: Missing type parameters for generic type "DiGraph"  [type-arg]
tel2puml/node_map_to_puml/node.py:1100: error: Argument 1 to "get_node_as_list" has incompatible type "Node | None"; expected "Node"  [arg-type]
tests/node_map_to_puml/test_node.py:6: error: Skipping analyzing "pm4py": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tests/node_map_to_puml/test_node.py:32: error: Module "tel2puml.puml_graph.graph" does not explicitly export attribute "PUMLOperatorNodes"  [attr-defined]
tests/node_map_to_puml/test_node.py:94: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:118: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:150: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:182: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:215: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:263: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:301: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:338: error: Argument 2 to "load_logic_into_list" of "Node" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:445: error: Argument 3 to "load_logic_tree_into_nodes" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:473: error: Argument 3 to "load_all_logic_trees_into_nodes" has incompatible type "str"; expected "Literal['incoming', 'outgoing']"  [arg-type]
tests/node_map_to_puml/test_node.py:490: error: Need type annotation for "graph"  [var-annotated]
tests/node_map_to_puml/test_node.py:516: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:547: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:553: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:568: error: Need type annotation for "expected_incoming_event_type_logic"  [var-annotated]
tests/node_map_to_puml/test_node.py:584: error: Unsupported right operand type for in ("object")  [operator]
tests/node_map_to_puml/test_node.py:593: error: Value of type "object" is not indexable  [index]
tests/node_map_to_puml/test_node.py:653: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:662: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:792: error: Argument 1 to "audit_event_sequences_to_network_x" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:798: error: Argument 1 to "update_all_connections_from_clustered_events" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/node_map_to_puml/test_node.py:819: error: Argument 1 to "rotate_path" of "LogicBlockHolder" has incompatible type "Node | None"; expected "Node"  [arg-type]
tests/node_map_to_puml/conftest.py:6: error: Skipping analyzing "pm4py": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tests/node_map_to_puml/conftest.py:7: error: Skipping analyzing "pm4py.objects.process_tree.obj": module is installed, but missing library stubs or py.typed marker  [import-untyped]
tel2puml/pv_to_puml.py:52: error: Argument 1 to "remove_loop_data_from_graph" has incompatible type "DiGraph[Any]"; expected "MultiDiGraph[Any]"  [arg-type]
tel2puml/pv_to_puml.py:55: error: Argument 3 to "remove_detected_loop_data_from_events" has incompatible type "dict[str, Event]"; expected "dict[str, str]"  [arg-type]
tel2puml/pv_to_puml.py:60: error: Argument 1 to "merge_markov_without_loops_and_logic_detection_analysis" has incompatible type "tuple[DiGraph[Any], dict[str, Event]]"; expected "tuple[DiGraph[Any], dict[str, str]]"  [arg-type]
tel2puml/pv_to_puml.py:110: error: Returning Any from function declared to return "list[PVEvent]"  [no-any-return]
tel2puml/node_map_to_puml/node_population_functions.py:11: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_population_functions.py:11: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/node_map_to_puml/node_population_functions.py:11: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_population_functions.py:30: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_population_functions.py:30: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/node_map_to_puml/node_population_functions.py:33: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_population_functions.py:100: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_population_functions.py:128: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_population_functions.py:129: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_population_functions.py:147: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/node_map_to_puml/node_population_functions.py:191: error: Need type annotation for "node_ref" (hint: "node_ref: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/node_map_to_puml/node_population_functions.py:208: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tel2puml/node_map_to_puml/node_map_to_puml.py:9: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:9: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:10: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:51: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:53: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:54: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:105: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:106: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:124: error: Need type annotation for "node_tree" (hint: "node_tree: List[<type>] = ...")  [var-annotated]
tel2puml/node_map_to_puml/node_map_to_puml.py:145: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:146: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:146: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:189: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:190: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:190: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:239: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:239: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:239: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:276: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:276: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:276: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:339: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:340: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:361: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:361: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:382: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:385: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:386: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:387: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:388: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:440: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:440: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:442: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:443: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:479: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:481: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:482: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:483: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:557: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:559: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:560: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:561: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:633: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:635: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:636: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:637: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:637: error: Incompatible default for argument "output" (default has type "None", argument has type "list[Any]")  [assignment]
tel2puml/node_map_to_puml/node_map_to_puml.py:637: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
tel2puml/node_map_to_puml/node_map_to_puml.py:637: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
tel2puml/node_map_to_puml/node_map_to_puml.py:713: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:713: error: Function is missing a type annotation for one or more arguments  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:713: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:733: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:734: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:735: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:738: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:789: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:790: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:815: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:816: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:818: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:889: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:890: error: Missing type parameters for generic type "list"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:890: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:915: error: Argument 4 to "insert_item_using_property_key" has incompatible type "*list[object]"; expected "str"  [arg-type]
tel2puml/node_map_to_puml/node_map_to_puml.py:915: error: Argument 4 to "insert_item_using_property_key" has incompatible type "*list[object]"; expected "bool"  [arg-type]
tel2puml/node_map_to_puml/node_map_to_puml.py:923: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:924: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:948: error: Function is missing a return type annotation  [no-untyped-def]
tel2puml/node_map_to_puml/node_map_to_puml.py:949: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:951: error: Missing type parameters for generic type "dict"  [type-arg]
tel2puml/node_map_to_puml/node_map_to_puml.py:953: error: Missing type parameters for generic type "dict"  [type-arg]
tests/test_pv_to_puml.py:16: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_pv_to_puml.py:26: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_pv_to_puml.py:37: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_pv_to_puml.py:47: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
tests/test_node_population_functions.py:26: error: Need type annotation for "graph"  [var-annotated]
tests/test_node_population_functions.py:115: error: Argument 2 to "create_event_node_ref" has incompatible type "dict[str, list[str]]"; expected "dict[str, str]"  [arg-type]
tests/test_node_map_to_puml.py:106: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:118: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:130: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:165: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:180: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:195: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:208: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:240: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:285: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:301: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:339: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:342: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:346: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:349: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:353: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:356: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:360: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:363: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:367: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:370: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:405: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:408: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:412: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:415: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:419: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:422: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:426: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:429: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:433: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:436: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:440: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:443: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:467: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:468: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:471: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:472: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:475: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:478: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:482: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:485: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:489: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:492: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:496: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:499: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:506: error: Argument 2 to "get_tree_distance_from_convergence" has incompatible type "object"; expected "list[Any]"  [arg-type]
tests/test_node_map_to_puml.py:526: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:560: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:580: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:614: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:636: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:675: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:703: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:736: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:760: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:796: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:815: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:825: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:856: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:885: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:912: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:961: error: Need type annotation for "reverse_node_trees" (hint: "reverse_node_trees: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1018: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1045: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:1082: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1111: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:1154: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1183: error: Value of type "object" is not indexable  [index]
tests/test_node_map_to_puml.py:1221: error: Argument 3 to "handle_divergent_tree_children" has incompatible type "str"; expected "list[Any]"  [arg-type]
tests/test_node_map_to_puml.py:1241: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1268: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1307: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1361: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1399: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1401: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1418: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1442: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1465: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1489: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1521: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1544: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1575: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1606: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1636: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1659: error: Need type annotation for "output" (hint: "output: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1661: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1710: error: Need type annotation for "dictionary" (hint: "dictionary: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1719: error: Need type annotation for "dictionary"  [var-annotated]
tests/test_node_map_to_puml.py:1738: error: Need type annotation for "event_reference" (hint: "event_reference: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1774: error: Need type annotation for "event_reference" (hint: "event_reference: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1792: error: Need type annotation for "event_reference" (hint: "event_reference: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1810: error: Need type annotation for "event_reference" (hint: "event_reference: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1828: error: Need type annotation for "event_reference" (hint: "event_reference: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1854: error: Need type annotation for "event_reference" (hint: "event_reference: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1949: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1966: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:1980: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:1982: error: Incompatible types in assignment (expression has type "Node", variable has type "list[Node]")  [assignment]
tests/test_node_map_to_puml.py:2024: error: Argument 3 to "insert_item_using_property_key" has incompatible type "str"; expected "dict[Any, Any]"  [arg-type]
tests/test_node_map_to_puml.py:2024: error: Argument 4 to "insert_item_using_property_key" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2033: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2049: error: Need type annotation for "missing_nodes" (hint: "missing_nodes: List[<type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2050: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2085: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2127: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2169: error: Need type annotation for "logic_lines" (hint: "logic_lines: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2196: error: Need type annotation for "logic_table" (hint: "logic_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2215: error: Need type annotation for "lookup_table" (hint: "lookup_table: Dict[<type>, <type>] = ...")  [var-annotated]
tests/test_node_map_to_puml.py:2252: error: Argument 1 to "len" has incompatible type "object"; expected "Sized"  [arg-type]
tests/test_node_map_to_puml.py:2253: error: Need type annotation for "item"  [var-annotated]
tests/test_node_map_to_puml.py:2254: error: Argument 1 to "enumerate" has incompatible type "object"; expected "Iterable[Never]"  [arg-type]
tests/test_node_map_to_puml.py:2260: error: Argument 1 to "len" has incompatible type "object"; expected "Sized"  [arg-type]
tests/test_node_map_to_puml.py:2262: error: Argument 1 to "enumerate" has incompatible type "object"; expected "Iterable[Never]"  [arg-type]
tests/test_node_map_to_puml.py:2344: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:2394: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:2461: error: Argument "incoming" to "Node" has incompatible type "Node"; expected "list[Node] | None"  [arg-type]
tests/test_node_map_to_puml.py:2509: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2563: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2625: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2684: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2772: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2839: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2903: error: Argument 2 to "process_puml_into_graphs" has incompatible type "bool"; expected "str"  [arg-type]
tests/test_node_map_to_puml.py:2913: error: Need type annotation for "logic_table" (hint: "logic_table: Dict[<type>, <type>] = ...")  [var-annotated]
tel2puml/utils_test.py:33: error: Argument 1 to "pv_to_puml_string" has incompatible type "Generator[Generator[dict[Any, Any], Any, None], Any, None]"; expected "Iterable[Iterable[PVEvent]]"  [arg-type]
Found 511 errors in 30 files (checked 43 source files)
